Database connection established!
Authentication established!
Httpreq: 
HttpRequest HTTP/1.1 GET:/api/auth/callback
  query: ?"code=b6062b4436ca54175902a7314b5244588ff1a98b1d8339664157009d7dd395f9&state=P3osAyUKA9Zo9tEuMmWMvA"
  headers:
    "accept-language": "en-DE,en;q=0.9,de-DE;q=0.8,de;q=0.7,en-GB;q=0.6,en-US;q=0.5"
    "accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7"
    "sec-ch-ua-platform": "\"macOS\""
    "sec-fetch-user": "?1"
    "sec-ch-ua": "\"Not.A/Brand\";v=\"8\", \"Chromium\";v=\"114\", \"Google Chrome\";v=\"114\""
    "upgrade-insecure-requests": "1"
    "sec-fetch-dest": "document"
    "host": "127.0.0.1:8080"
    "connection": "keep-alive"
    "accept-encoding": "gzip, deflate, br"
    "user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"
    "cache-control": "max-age=0"
    "sec-ch-ua-mobile": "?0"
    "sec-fetch-mode": "navigate"
    "sec-fetch-site": "cross-site"

client: Data(Client { client_id: ClientId("u-s4t2ud-ef501d4b06f023833eb0ddba051b80feb2e7e79f8805444e8cf074a32a8d5f67"), client_secret: Some(ClientSecret([redacted])), auth_url: AuthUrl("https://api.intra.42.fr/oauth/authorize"), auth_type: BasicAuth, token_url: Some(TokenUrl("https://api.intra.42.fr/oauth/token")), redirect_url: Some(RedirectUrl("http://127.0.0.1:8080/api/auth/callback")), introspection_url: None, revocation_url: None, device_authorization_url: None, phantom: PhantomData<(oauth2::StandardErrorResponse<oauth2::basic::BasicErrorResponseType>, oauth2::StandardTokenResponse<oauth2::EmptyExtraTokenFields, oauth2::basic::BasicTokenType>, oauth2::basic::BasicTokenType, oauth2::StandardTokenIntrospectionResponse<oauth2::EmptyExtraTokenFields, oauth2::basic::BasicTokenType>, oauth2::revocation::StandardRevocableToken, oauth2::StandardErrorResponse<oauth2::revocation::RevocationErrorResponseType>)> })
Httpreq: 
HttpRequest HTTP/1.1 GET:/api/auth/callback
  query: ?"error=access_denied&error_description=The+resource+owner+or+authorization+server+denied+the+request.&state=P3osAyUKA9Zo9tEuMmWMvA"
  headers:
    "host": "127.0.0.1:8080"
    "sec-fetch-site": "cross-site"
    "sec-fetch-user": "?1"
    "cache-control": "max-age=0"
    "user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"
    "sec-ch-ua-platform": "\"macOS\""
    "accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7"
    "sec-ch-ua-mobile": "?0"
    "accept-language": "en-DE,en;q=0.9,de-DE;q=0.8,de;q=0.7,en-GB;q=0.6,en-US;q=0.5"
    "sec-fetch-dest": "document"
    "sec-ch-ua": "\"Not.A/Brand\";v=\"8\", \"Chromium\";v=\"114\", \"Google Chrome\";v=\"114\""
    "accept-encoding": "gzip, deflate, br"
    "upgrade-insecure-requests": "1"
    "sec-fetch-mode": "navigate"
    "connection": "keep-alive"

client: Data(Client { client_id: ClientId("u-s4t2ud-ef501d4b06f023833eb0ddba051b80feb2e7e79f8805444e8cf074a32a8d5f67"), client_secret: Some(ClientSecret([redacted])), auth_url: AuthUrl("https://api.intra.42.fr/oauth/authorize"), auth_type: BasicAuth, token_url: Some(TokenUrl("https://api.intra.42.fr/oauth/token")), redirect_url: Some(RedirectUrl("http://127.0.0.1:8080/api/auth/callback")), introspection_url: None, revocation_url: None, device_authorization_url: None, phantom: PhantomData<(oauth2::StandardErrorResponse<oauth2::basic::BasicErrorResponseType>, oauth2::StandardTokenResponse<oauth2::EmptyExtraTokenFields, oauth2::basic::BasicTokenType>, oauth2::basic::BasicTokenType, oauth2::StandardTokenIntrospectionResponse<oauth2::EmptyExtraTokenFields, oauth2::basic::BasicTokenType>, oauth2::revocation::StandardRevocableToken, oauth2::StandardErrorResponse<oauth2::revocation::RevocationErrorResponseType>)> })